internal/lldp/neigh.go

```diff
@@ -1,27 +1,47 @@
 package lldp
 
-import "time"
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+)
+
+type ManagementAddress struct {
+	AddressFamily    string `json:"address_family"`
+	Address          string `json:"address"`
+	InterfaceSubtype string `json:"interface_subtype"`
+	InterfaceNumber  uint32 `json:"interface_number"`
+	OID              string `json:"oid,omitempty"`
+}
 
 type Neighbor struct {
-	Mac               string            `json:"mac"`
-	Source            string            `json:"source"`
-	ChassisID         string            `json:"chassis_id"`
-	PortID            string            `json:"port_id"`
-	PortDescription   string            `json:"port_description"`
-	SystemName        string            `json:"system_name"`
-	SystemDescription string            `json:"system_description"`
-	TTL               uint16            `json:"ttl"`
-	ManagementAddress string            `json:"management_address"`
-	Values            map[string]string `json:"values"`
+	Mac               string             `json:"mac"`
+	Source            string             `json:"source"`
+	ChassisID         string             `json:"chassis_id"`
+	PortID            string             `json:"port_id"`
+	PortDescription   string             `json:"port_description"`
+	SystemName        string             `json:"system_name"`
+	SystemDescription string             `json:"system_description"`
+	TTL               uint16             `json:"ttl"`
+	ManagementAddress *ManagementAddress `json:"management_address,omitempty"`
+	Capabilities      []string           `json:"capabilities"`
+	Values            map[string]string  `json:"values"`
+}
+
+func (n *Neighbor) cacheKey() string {
```

Comment:
This does a Sprintf every time, much cheaper to do this, which is almost zero-allocating:

```go
type NeighborCacheKey struct {
    Mac    string
    Source string
}

func (n *Neighbor) cacheKey() NeighborCacheKey {
   return NeighborCacheKey {Mac: n.Mac, Source: item.Source}
}
```